var/spack/repos/builtin/packages/python/package.py

```diff
@@ -432,11 +432,23 @@ def libs(self):
         library   = self.get_config_var('LIBRARY')
         ldlibrary = self.get_config_var('LDLIBRARY')
 
+        # MacOS/X Frameworks throw a spanner in the soup
+        libdir_prefix = os.path.dirname(libdir)
+        ldlibrary_dir = os.path.dirname(ldlibrary)
+        ldlibrary_name = os.path.basename(ldlibrary)
+        framework_library = ""
+        if len(libdir_prefix) and len(ldlibrary_dir) and len(ldlibrary_name) \
+           and libdir_prefix[-len(ldlibrary_dir):] == ldlibrary_dir \
+           and os.path.exists(os.path.join(libdir_prefix, ldlibrary_name)):
+            framework_ldlibrary = os.path.join(libdir_prefix, ldlibrary_name)
```

Comment:
I see what you're trying to do here, but I think there's a simpler way. It looks like the library can be found using `PYTHONFRAMEWORKPREFIX` + `LDLIBRARY`. That key seems to exist even on Linux (although it's empty), so it should be easy to check for the presence of either `LIBDIR` + `LDLIBRARY` or `PYTHONFRAMEWORKPREFIX` + `LDLIBRARY`. I'll put out a PR shortly for you to test.